power_grid_model_c/power_grid_model/include/power_grid_model/calculation_parameters.hpp

```diff
@@ -92,7 +93,7 @@ struct MathModelTopology {
     std::vector<double> phase_shift;
     std::vector<BranchIdx> branch_bus_idx;
     std::vector<BranchIdx> fill_in;
-    IdxVector source_bus_indptr;
+    SparseGroupedIdxVector source_buses; // TODO(mgovers) replace with DenseGroupedIdxVector
```

Comment:
maybe we want to make abstraction of whether it is sparse or dense. see also

https://github.com/PowerGridModel/power-grid-model/pull/379#discussion_r1354195358

Reply:
I am not in favor of determine dense/sparse mapping dynamically. It incurs runtime cost. We need to use power system prior knowledge to determine which mapping is dense or sparse at compile time.

Reply:
ok fair enough.